### PR TITLE
Test with default CLI opts

### LINKED ISSUE #2252: Add end-to-end tests: add CI importing documents to the Meilisearch server
To prevent having this [kind of issue](https://github.com/meilisearch/meilisearch/issues/2251) in the future, we need to add a CI that checks the Meilisearch server can receive documents and well process them.

A basic version of the CI script should
- Run the Meilisearch server without any argument
- Add a simple document, ex: `[{"id": 1, "name": "toto l'asticot"}]`
- Check the document was well processed.

The script should be run in a new job in [this file](https://github.com/meilisearch/meilisearch/blob/main/.github/workflows/rust.yml).
We also need to add this job to our [bors.toml](https://github.com/meilisearch/meilisearch/blob/main/bors.toml) file.

### PR DESCRIPTION
**Pull request by Kerollmops**
Fixes #2252.

This PR makes sure that we test the HTTP engine with the default CLI parameters and removes some useless internal CLI options.

### CODE REVIEW

**Comment by MarinPostma** [comment]
@Kerollmops tests are failing, clap seems to need a default implementation for IndexerOpts

**Comment by Kerollmops** [comment]
Hum... I tested that on another branch where the `indexer_opts` argument is not skipped and it worked fine, obviously. I will implement `Default` on this struct again as I removed the problematic options and it looks safe now.

**Comment by MarinPostma** [comment]
@Kerollmops I have concerns that this will break windows tests: we need 100gib map for each running test thread

**Comment by Kerollmops** [comment]
Wat do you mean? I kept the `max_index_size` limits here.

**Comment by MarinPostma** [comment]
my bad! test still failed with OOM though

File: meilisearch-lib/src/options.rs [hunk 1]
```diff
@@ -91,13 +78,11 @@ impl TryFrom<&IndexerOpts> for IndexerConfig {
 }
 
 impl Default for IndexerOpts {
-    fn default() -> Self {
-        Self {
+    fn default() -> IndexerOpts {
+        IndexerOpts {
```
Comment (ManyTheFish) [inline meilisearch-lib/src/options.rs:82]: Why did you choose to explicitly specify the name?

**Review by ManyTheFish — CHANGES_REQUESTED** [review]

**Review by ManyTheFish — APPROVED** [review]

**Comment by Kerollmops** [comment]
We need to rebase this branch on #2280 to avoid this issue with lindera.

**Comment by Kerollmops** [comment]
I just rebased on #2280, you can re-review this PR and we will be able to merge as the CI should compile now.

**Review by MarinPostma — APPROVED** [review]
Thanks @Kerollmops ! Happy to be able to merge that now 😁

**Comment by Kerollmops** [comment]
Thank you!
bors merge

**Pull request merged**